apis/python/src/tiledbsoma/common_nd_array.py

```diff
@@ -3,15 +3,54 @@
 TODO: Extract more stuff.
 """
 
-from typing import Sequence
+from typing import Optional, Sequence, Type, TypeVar
 
 import numpy as np
 import pyarrow as pa
+import somacore
 import tiledb
+from somacore import options
 
 from . import util, util_arrow
 from .options.soma_tiledb_context import SOMATileDBContext
 from .options.tiledb_create_options import TileDBCreateOptions
+from .tiledb_array import TileDBArray
+
+_Self = TypeVar("_Self", bound="NDArray")
+
+
+class NDArray(TileDBArray, somacore.NDArray):
```

Comment:
could this be documented as being a base class and is useful for polymorphic typing?

Reply:
Added a brief note. If you think it needs a little more context, please do say so.